app/models.py

```diff
@@ -0,0 +1,17 @@
+import typing as t
+
+from pydantic import BaseModel
+
+
+class AdminTable(BaseModel):
+    table_name: t.Optional[str] = None
+    visible_columns: t.Optional[list[str]] = None
+    visible_filters: t.Optional[list[str]] = None
+    rich_text_columns: t.Optional[str] = None
+    link_column: t.Optional[str] = None
+    menu_group: t.Optional[str] = None
+
+
+class AdditionalConfig(BaseModel):
+    tables: t.Optional[list[AdminTable]] = None
```

Comment:
My only suggestion here would be to have string as an option, and rather than defaulting to None, default to an empty list.

Something like:

```python
tables: list[AdminTable | str] = field(default=list)
```

So if the user just wants to show a table, and isn't bothered about passing in any config options, we just show the tables with a default config.

If no tables are specified, so we default to showing the user all of their tables?

Reply:
I think we should stick with `tables: list[AdminTable] | None = None` as we cover all options with little change.
Option 1:
The user does not pass anything to `yaml` file, something like this
```yaml
tables:
  # - table_name: actor
  #   visible_columns:
  #     - first_name
  #   visible_filters:
  #     - actor_id
  #     - first_name
  #   menu_group: Movies
  #   link_column: first_name
```
We return the table with default columns, etc.

Option 2:
The user pass everything (or only `visible_columns` for example) to `yaml` file, something like this
```yaml
tables:
  - table_name: actor
    visible_columns:
      - first_name
    visible_filters:
      - actor_id
      - first_name
    menu_group: Movies
    link_column: first_name
```
We return the table with specified configuration.

Option 3:
The user pass only `table_name` to `yaml` file, something like this
```yaml
tables:
  - table_name: actor
```
We need to make a small change in the visible columns and filters so that we return the default table configuration if `TypeError` (for `NoneType`) is raised. After that change everything works as in `Option 1`.

Reply:
OK, fair enough - I'm happy if option 3 works, thanks.

Reply:
I'm going to make that little change right away. Thank you for your review and time.

Reply:
No problem, thanks!